plugin/federation/testdata/entityresolver/generated/federation.go

```diff
@@ -0,0 +1,132 @@
+// Code generated by github.com/99designs/gqlgen, DO NOT EDIT.
+
+package generated
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/99designs/gqlgen/plugin/federation/fedruntime"
+)
+
+func (ec *executionContext) __resolve__service(ctx context.Context) (fedruntime.Service, error) {
+	if ec.DisableIntrospection {
+		return fedruntime.Service{}, errors.New("federated introspection disabled")
+	}
+
+	var sdl []string
+
+	for _, src := range sources {
+		if src.BuiltIn {
+			continue
+		}
+		sdl = append(sdl, src.Input)
+	}
+
+	return fedruntime.Service{
+		SDL: strings.Join(sdl, "\n"),
+	}, nil
+}
+
+func (ec *executionContext) __resolve_entities(ctx context.Context, representations []map[string]interface{}) []fedruntime.Entity {
+	list := make([]fedruntime.Entity, len(representations))
+	resolveEntity := func(ctx context.Context, i int, rep map[string]interface{}) (err error) {
+		// we need to do our own panic handling, because we may be called in a
+		// goroutine, where the usual panic handling can't catch us
+		defer func() {
+			if r := recover(); r != nil {
+				err = ec.Recover(ctx, r)
+			}
+		}()
+
+		typeName, ok := rep["__typename"].(string)
+		if !ok {
+			return errors.New("__typename must be an existing string")
+		}
+		switch typeName {
+		case "Hello":
+			entity, err := func() (*generated.Hello, error) {
+				id0, err := ec.unmarshalNString2string(ctx, rep["name"])
+				if err == nil {
+					return ec.resolvers.Entity().FindHelloByName(ctx, id0)
+				}
+				return nil, nil
+			}()
+
+			if err != nil {
+				return fmt.Errorf(`resolving Entity "Hello": %w`, err)
+			}
+			if entity == nil {
+				return errors.New(`unable to resolve Entity "Hello"`)
+			}
+
+			list[i] = entity
+			return nil
+
+		case "World":
+			entity, err := func() (*generated.World, error) {
+				id0, err := ec.unmarshalNString2string(ctx, rep["hello"].(map[string]interface{})["name"])
```

Comment:
This is another bug if the entity query is for the key `bar` then this will throw an error.